GHzDACs/jumpTable.py

```diff
@@ -4,182 +4,234 @@
-    jumpIndex = property(getJumpIndex, setJumpIndex)
-    
+            raise ValueError("Must have {0} < jump index < {1}, not {2}".format(JUMP_INDEX_MIN, JUMP_INDEX_MAX, idx))
+
     def __str__(self):
-        raise NotImplementedError
-    
-    def asBytes(self):
+        raise NotImplementedError()
+
+    def as_bytes(self):
         """Override in subclass"""
-        raise NotImplementedError
+        raise NotImplementedError()
+
 
 class IDLE(Operation):
-    """Wraps the IDLE jump table op code"""
-    
     NAME = "IDLE"
-    
+
     def __init__(self, cycles):
-        self._cycles = cycles
-    
-    def getJumpIndex(self):
-        raise RuntimeError('IDLE does not support jump table indexing')
-    def setJumpIndex(self, idx):
-        raise RuntimeError('IDLE does not support jump table indexing')
-    
-    def getCycles(self):
+        self._cycles = 0
+        self.cycles = cycles - 1  # account for FPGA firmware timing offset
+
+    @property
+    def cycles(self):
         return self._cycles
-    def setCycles(self, cycles):
-        if IDLE_MIN_CYCLES < cycles and cycles < IDLE_MAX_CYCLES:
+
+    @cycles.setter
+    def cycles(self, cycles):
+        if IDLE_MIN_CYCLES < cycles < IDLE_MAX_CYCLES:
             self._cycles = cycles
         else:
-            raise RuntimeError('Number of idle cycles must satisfy: %s < cycles < %s'%(IDLE_MIN_CYCLES,IDLE_MAX_CYCLES))
-    cycles = property(getCycles, setCycles)
-    
+            raise ValueError(
+                'Number of idle cycles must satisfy: {0} < cycles < {1}, '
+                'not {2}'.format(IDLE_MIN_CYCLES, IDLE_MAX_CYCLES, cycles)
+            )
+
     def __str__(self):
-        return "%s %d cycles"%(self.NAME, self.cycles)
-    
-    def asBytes(self):
-        return self._cycles<<1
-    
-class CHECK(Operation):
+        return "%s %d cycles" % (self.NAME, self.cycles)
 
+    def as_bytes(self):
+        return self._cycles << 1
+
+
+class CHECK(Operation):
     NAME = "CHECK"
-    
-    def __init__(self, whichDaisyBit, bitOnOff, nextJumpIndex):
-        raise RuntimeError('CHECK is not yet understood')
-        self.whichDaisyBit = whichDaisyBit
-        self.jumpIndex = nextJumpIndex
-        self.bitOnOff = bool(bitOnOff)
-        
-    def getWhichDaisyBit(self):
-        return self._whichDaisyBit
-    def setWhichDaisyBit(self, whichBit):
-        if DAISY_VALUE_MIN < whichBit and whichBit < DAISY_VALUE_MAX:
-            self._whichDaisyBit = whichBit
+
+    def __init__(self, which_daisy_bit, bit_state, next_jump_index):
+        self._which_daisy_bit = 0
+        self.which_daisy_bit = which_daisy_bit
+        self.jump_index = next_jump_index
+        self.bitOnOff = bool(bit_state)
+        raise NotImplementedError('CHECK is not yet implemented')
+
+    @property
+    def which_daisy_bit(self):
+        return self._which_daisy_bit
+
+    @which_daisy_bit.setter
```

Comment:
It would be nice to encapsulate all this range checking logic into a descriptor class. Then you could write something like:
```python
@range_checked(min=DAISY_VALUE_MIN, max=DAISY_VALUE_MAX)
def which_daisy_bit(self):
    pass
```
and put all the getter/setter logic in the descriptor. I think this would cut down a lot of boilerplate and make the intent more clear.